UserDataProcessingTrigger/index.ts

```diff
@@ -1,20 +1,38 @@
 import { Context } from "@azure/functions";
-import { triggerHandler } from "./handler";
+import { createTableService } from "azure-storage";
+import { getConfigOrThrow } from "../utils/config";
 import {
-  addFailedUserDataProcessing,
-  removeFailedUserDataProcessing,
-  createFailedUserDataProcessingTableIfNotExists
-} from "./utils";
+  createTableIfNotExists,
+  deleteTableEntity,
+  insertTableEntity
+} from "../utils/storage";
+import { triggerHandler } from "./handler";
 
-createFailedUserDataProcessingTableIfNotExists();
+// configure log prefix
+const logPrefix = "UserDataProcessingTrigger";
 
-export const index = (
+// prepare table storage utils
+const config = getConfigOrThrow();
+const connectionString = config.FailedUserDataProcessingStorageConnection;
+const failedUserDataProcessingTable = config.FAILED_USER_DATA_PROCESSING_TABLE;
+const tableService = createTableService(connectionString);
+
+export const index = async (
   context: Context,
   input: unknown
 ): Promise<ReadonlyArray<string | void>> => {
+  (await createTableIfNotExists(
+    tableService,
+    failedUserDataProcessingTable
+  )).mapLeft(_ => {
+    context.log.verbose(
+      `${logPrefix}|Failed to create storage table ${failedUserDataProcessingTable}`
+    );
+  });
```

Comment:
Doing like this we will perform a useless request to the storage at every document inserted in the collection, while it's (maybe) needed only on first call.

I realize I was misleading in previous comment, as I advised for handling the possible failure of the operation. Maybe is not so important (it would be needed the first time only).

Thought?

Reply:
If we want to avoid repetition I think we have a couple of cases: 
1. creating it with platform project and terraform, or
2. creating it when we receive a "table not found error" during insert and delete operations, and retry them.

Reply:
Another option: use built-in bindings and let the runtime do the job for us:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-table-output?tabs=javascript
(you can use env var in `function.json` using `%VAR_NAME%` notation)

Reply:
Nice, let me check!

Reply:
Ok, checked. Well, it seems that these bindings allow us to query and insert data, and should also create table during the insert to context. So I think we can do an `insertToTable` utils that insert a record using the context and use it also elsewhere. Maybe I can refactor also the two functions that gets the data to use a `getFromTable` that uses context too. If you agree with this I can give it a try. From some internet blog posts it seems that this way is also more optimized to insert and read, missing just update and delete operations that could be done in the old way.